### PR TITLE
feat(tools): add file tool confinement controls [ST-11003]

### PR DESCRIPTION
**Pull request by TVScoundrel**
## Story

`ST-11003`: Add Filesystem Confinement Controls for Default File Tools

## What Changed

Added a shared `FileSystemPolicy` to the file and directory tool factories. The policy supports configured allowed roots or workspace-relative operation, rejects traversal and resolved symlink escapes, protects recursive deletion of configured roots, and preserves an explicit privileged opt-out for trusted local automation. Pure path utilities remain unchanged.

## Acceptance Criteria Coverage

- [x] Default file read and mutation tools support configurable confinement without bespoke wrappers.
- [x] Traversal, symlink escape, and destructive recursive-delete edge cases are covered by focused tests.
- [x] Privileged operator workflows retain a documented explicit opt-out.
- [x] Public docs distinguish model-exposed file tools from trusted local automation.
- [x] Story documentation is added at `docs/st11003-file-tool-confinement-controls.md`.

## Test Strategy

Red-first tests cover a shared policy resolver and configured file/directory factories using temporary roots and symlink fixtures. The suite validates workspace-relative paths, lexical traversal, resolved symlink containment, creation parents, recursive root deletion, safe directory listing, and the privileged opt-out.

## Validation Performed

Focused confinement suite: 8 tests passed. Tools package suite: 90 files passed, 9 skipped; 1153 tests passed, 110 skipped. Repo-wide `pnpm test --run` passed with 226 files passed, 9 skipped; 2534 tests passed, 110 skipped. `pnpm lint` passed with 0 errors and the existing 161-warning baseline. `pnpm typecheck` and `pnpm build` passed; build retained the existing VitePress chunk-size warning.

## Status

Ready for review. Implementation, documentation, focused and repo-wide validation, lint, typecheck, build, self-review, and tracker synchronization are complete.

### CODE REVIEW

File: packages/tools/src/file/operations/tools/file-exists.ts [hunk 1]
```diff
@@ -5,21 +5,23 @@
...
-export function createFileExistsTool() {
+export function createFileExistsTool(policy: FileSystemPolicy = DEFAULT_FILE_SYSTEM_POLICY) {
   return toolBuilder()
     .name('file-exists')
     .description('Check if a file or directory exists at the specified path.')
     .category(ToolCategory.FILE_SYSTEM)
     .tags(['file', 'exists', 'check', 'filesystem'])
     .schema(fileExistsSchema)
     .implement(async (input) => {
+      const safePath = await policy.resolvePath(input.path, 'file existence check');
```
Comment (TVScoundrel) [inline packages/tools/src/file/operations/tools/file-exists.ts]: `resolvePath` is called outside the `try/catch` on line 22, so `FileSystemPolicyError` propagates as an unhandled rejection instead of being caught and returned as `{ success: false, error }`. All other tools in this PR use `.implementSafe()`, which wraps the handler in a catch that converts errors to that shape. The test at `confinement.test.ts:146` explicitly asserts `.rejects.toMatchObject({ code: "FILE_SYSTEM_POLICY_VIOLATION" })`, confirming the behaviour is intentional — but the story doc (`docs/st11003-file-tool-confinement-controls.md` line 28) states "safe tool implementations expose those through their normal `{ success: false, error }` result shape", which is not true for this tool.

Either move `resolvePath` inside the try/catch (so a confinement violation returns `{ exists: false }` or a distinct error object rather than throwing), or explicitly call out the `file-exists` exception in the story doc and the compatibility section of the README. Leaving it as-is silently breaks any caller that switches to a confined policy and wraps all tool invocations with `{ success }` checking instead of rejection handling.

File: packages/tools/src/file/confinement.ts [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    }
+
+    return resolvedPath;
+  }
+
+  async resolveDeletePath(inputPath: string, recursive: boolean): Promise<string> {
+    const resolvedPath = await this.resolvePath(inputPath, 'directory deletion');
+
+    if (recursive && !this.allowRootDeletion) {
+      const resolvedRoots = await this.resolveRoots('directory deletion', inputPath);
```
Comment (TVScoundrel) [inline packages/tools/src/file/confinement.ts]: `resolveRoots` is called twice for every recursive delete: once inside `resolvePath` (line 106) and once here explicitly. Both calls do `Promise.all(this.allowedRoots.map(root => fs.realpath(root)))`, so the cost is doubled. For most setups with one or two roots this is negligible, but it is unnecessary work. Consider extracting the resolved roots from a single call, or having `resolvePath` return them alongside the resolved path so `resolveDeletePath` can reuse the result.

File: packages/tools/src/file/directory/tools/directory-list.ts [hunk 3]
```diff
@@ -33,8 +40,8 @@ export function createDirectoryListTool(defaultRecursive: boolean = false, defau
             }
           }
 
-          if (input.includeDetails) {
-            const stats = await fs.stat(fullPath);
+          if (includeDetails) {
+            const stats = await fs.lstat(fullPath);
```
Comment (TVScoundrel) [inline packages/tools/src/file/directory/tools/directory-list.ts:44]: Changing from `fs.stat` to `fs.lstat` is the right call for the confinement use-case (symlinks to outside-root directories should not masquerade as directories), but this change applies to **all** callers — including those using the unconfigured default policy. Before this PR, `directory-list` followed symlinks for stats, so a symlink to a directory reported `isDirectory: true`. After this PR it reports `isDirectory: false` for all callers, even legacy unconfigured ones that never opted into any policy.

This is a silent behavioral breaking change. The test documents the intent ("non-following directory-list symlinks"), but the Compatibility section of `docs/st11003-file-tool-confinement-controls.md` only mentions the additive `policy` option and does not call out the `lstat` change. Consider adding a note there so upgraders know that `isDirectory` will now be `false` for directory symlinks in all listing results.
